### DISCUSSION TITLE
How to allow via generation of the top routing layer

### DISCUSSION THREAD

**Discussion by b224hisl** [The-OpenROAD-Project/OpenROAD-flow-scripts#2062] [Q&A]
Excuse, I have noticed during the operation of ORFS that if the `MAX_ROUTING_LAYER` is set, it will not generate vias on that layer. If I want to route wires on this layer but generate vias, what should I do?

**Comment by maliberty** [comment]
You should get vias from below but not from above (as this is the max).

**Reply by b224hisl** [reply]
I think it will get vias from the below layer. Because when I set the `MAX_ROUTING_LAYER = Metal4`, It will report:
```
Up-via summary (total  67810):
----------------
  Poly2        0
 Metal1    44120
 Metal2    22105
 Metal3     1585
 Metal4     0
----------------
           67810
```
when do the detailed routing

**Reply by maliberty** [reply]
That looks right so I'm not sure what your concern/question is.
